Add gitleaks cli command and enable gitleaks detect scans from the report command (#95)
RedHatQE/firewatch: docs/cli_usage_guide.md

**Arguments:**

```commandline
Usage: firewatch gitleaks [OPTIONS]

Options:
  --keep-job-dir      If set, firewatch will not delete the job directory
                      (/tmp/12345) that is created to hold logs and results
                      for a job following execution.
  --output-file PATH  The name of the gitleaks detect report to be stored as
                      an artifact
  --token-path PATH   Path to the Redhat patterns server token  [required]
  --server-url TEXT   Redhat patterns server URL  [required]
  --help              Show this message and exit.
```

**Examples:**

```commandline
# Run a Firewatch report command without deleting the job directory in /tmp, which will be scanned by Gitleaks.
$ firewatch report --keep-job-dir

# Perform a Gitleaks detect scan with the default expected paths and server URL, and remove the job directory upon completion.

> $ firewatch gitleaks

Reviewer: Will this firewatch gitleaks command lead to a jira ticket being created so that the security issue can get worked on and escalated?

In this flow gitleaks is running after firewatch report, which makes me think a ticket wouldn't be created for a high priority security issue.

Is there anything preventing us from flipping this?
1. Run gitleaks
2. Report issues

I guess the run gitleaks part would need to do the downloading of the files in that case.

Also what do we think about just making gitleaks an option to the firewatch report command?
- firewatch report --git-leaks true (The intent here would be to make sure any time we run a scenario firewatch creates tickets for security issues).

Reviewer 2: @ascerra We can't flip it because report needs to download everything and I want to keep it that way as gitleaks should be a secondary functionality to firewatch. 

As for adding it as an option, I think it may muddy the waters too much. I think keeping it as its own command is fine. Authenticating to Jira is easy and integrating with all the Jira functionality I have is also easy. It's just a matter of adding logic to detect when that happens and reporting on it.

Reviewer: Ack. Thanks for sharing some details I think allowing the standalone firewatch gitleaks command to report to Jira would make sense to me.

I can't think of a reason it needs to be forced into the report command after reading what you shared.

# Create a configuration file in a different location (/some/path/jira.config)
$ firewatch jira-config-gen --token-path {Path to file containing Jira API token} --server-url https://some.jira.server.com --output-file /some/path
```

### `gitleaks`

The `gitleaks` command is used to run a Gitleaks detect scan on the job directory created by running the [`report`](#report) command.

> The latest Redhat detection rules are fetched from the [pattern distribution server](https://source.redhat.com/departments/it/it-information-security/wiki/pattern_distribution_server) and used by default.

Reviewer 2: ```suggestion
The latest Red Hat detection rules are fetched from the [pattern distribution server](https://source.redhat.com/departments/it/it-information-security/wiki/pattern_distribution_server) and used by default.
```

This command is intended to be run following a [`report`](#report) command that has the `--keep-job-dir` flag set.

**Pre-requisites:**

1. The job directory created by the [`report`](#report) command, which contains the downloaded artifacts to be scanned. By default, the job directory is removed unless the `--keep-job-dir` flag is set.
2. A patterns distribution server API token. By default, the file containing the token is expected to be mounted as a secret in Openshift CI at `/tmp/secrets/rh-patterns-server/access-token`. See the [pattern distribution server Source page](https://source.redhat.com/departments/it/it-information-security/wiki/pattern_distribution_server) for more information.

**Arguments:**

```commandline
Usage: firewatch gitleaks [OPTIONS]

Options:
  --keep-job-dir      If set, firewatch will not delete the job directory
                      (/tmp/12345) that is created to hold logs and results
                      for a job following execution.
  --output-file PATH  The name of the gitleaks detect report to be stored as
                      an artifact

>   --token-path PATH   Path to the Redhat patterns server token  [required]

Reviewer 2: ```suggestion
  --token-path PATH   Path to the Red Hat patterns server token  [required]
```

>   --server-url TEXT   Redhat patterns server URL  [required]

Reviewer 2: ```suggestion
  --server-url TEXT   Red Hat patterns server URL  [required]
```
